src/core/thread/mle_router.hpp

```diff
@@ -552,9 +552,12 @@ class MleRouter: public Mle
     bool HandleAdvertiseTimer(void);
     static void HandleStateUpdateTimer(void *aContext);
     void HandleStateUpdateTimer(void);
+    static void HandleChallengeInvalidateTimer(void *aContext);
+    void HandleChallengeInvalidateTimer(void);
 
     TrickleTimer mAdvertiseTimer;
     Timer mStateUpdateTimer;
+    Timer mChallengeInvalidateTimer;
```

Comment:
We could avoid adding a new Timer object by just using a countdown variable that is decremented in `MleRouter::HandleStateUpdateTimer()` .